readchar/_posix_read.py

```diff
@@ -1,9 +1,68 @@
 import sys
 import termios
+from copy import copy
+from select import select
 
 from ._config import config
 
 
+class ReadChar:
+    """A ContextManager allowing for keypress collection without requiering the user to
+    confirm presses with ENTER. Can be used non-blocking while inside the context."""
+
+    def __init__(self, cfg: config = None) -> None:
+        self.config = cfg if cfg is not None else config
+
+    def __enter__(self) -> "ReadChar":
+        self.fd = sys.stdin.fileno()
+        term = termios.tcgetattr(self.fd)
+        self.old_settings = copy(term)
+        term[3] &= ~(termios.ICANON | termios.ECHO | termios.IGNBRK | termios.BRKINT)
+        termios.tcsetattr(self.fd, termios.TCSAFLUSH, term)
+        return self
+
+    def __exit__(self, type, value, traceback) -> None:
+        termios.tcsetattr(self.fd, termios.TCSADRAIN, self.old_settings)
+
+    @property
+    def key_waiting(self) -> bool:
```

Comment:
this currently isn't working consistently

Reply:
fixed in the lates commit.

Now `stdin` is read completly whenever `sys.stdin in select([sys.stdin], [], [], 0)[0]` returnes `True` and put into a memory-buffer. There it can be completly controlled, making peeking and reading much easier.